components/brave_wallet/common/zcash_utils.cc

```diff
@@ -208,31 +220,117 @@ std::optional<std::string> ExtractTransparentPart(
     return std::nullopt;
   }
 
-  std::vector<uint8_t> padded_hrp(kPaddedHrpSize, 0);
-  std::copy(reinterpret_cast<const uint8_t*>(expected_hrp.c_str()),
-            reinterpret_cast<const uint8_t*>(expected_hrp.c_str()) +
-                expected_hrp.length(),
-            padded_hrp.begin());
+  auto padded_hrp = GetPaddedHRP(is_testnet);
 
   // Check that HRP is similar to the padded HRP
   if (!std::equal(padded_hrp.begin(), padded_hrp.end(),
                   reverted->end() - kPaddedHrpSize)) {
     return std::nullopt;
   }
 
-  auto parts = ParseUnifiedAddress(
+  auto parts = ParseUnifiedAddressBody(
       base::make_span(*reverted).subspan(0, reverted->size() - kPaddedHrpSize));
+
+  return parts;
+}
+
+// https://zips.z.cash/zip-0316#encoding-of-unified-addresses
+std::optional<std::string> ExtractTransparentPart(
+    const std::string& unified_address,
+    bool is_testnet) {
+  auto transparent_bytes = GetTransparentRawBytes(unified_address, is_testnet);
```

Comment:
imo using exact types is more readable, b/c in the code like this string vs array vs vector vs anything else is important